x-pack/plugins/ml/public/ui_actions/open_in_anomaly_explorer_action.tsx

```diff
@@ -50,6 +51,33 @@ export function isAnomalyChartsEmbeddableContext(arg: unknown): arg is {
   );
 }
 
+// This is useful because embeddables do not differentiate between jobIds and groupIds
```

Comment:
@darnautov do you remember the context around why embeddables don't work with the concept of job groups? Could this behavior be changed (not in this PR) to pass the group ID rather than the IDs of all the jobs in the group? Just wondering why it works this way.

Reply:
Related - I guess this is why this message refers to 'no known job' even though when I created the embeddable I configured it with a job group (which I then deleted after creating the embeddable).

<img width="467" alt="Screenshot 2024-10-30 at 12 34 00" src="https://github.com/user-attachments/assets/2f178ab6-a575-4ef8-bb79-f1f11d1b9ac6">

Reply:
I recall we decided to do that because the ML backend has unified support for `job_id` as both job IDs and group names, e.g check https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-get-job.html

> <job_id>
> (Optional, string) Identifier for the anomaly detection job. It can be a job identifier, a group name, or a wildcard expression.

The other approach to fix this issue is changing the URL state code in Anomaly Explorer to support group names alongside job IDs. I assumed it'd be a bit more work,  but it'd be better for reducing API calls. 

The current fix requires 2 additional API calls from the UI action code, but the Anomaly Explorer should have all necessary information upfront.

Reply:
> The other approach to fix this issue is changing the URL state code in Anomaly Explorer to support group names alongside job IDs. I assumed it'd be a bit more work, but it'd be better for reducing API calls.

This might be worth doing in the future, but agree that we shouldn't do it in this PR to avoid the additional code churn for this fix.

Reply:
A high-level overview of an alternative fix: 
1. Perform only one API call `getAllJobAndGroupIds` to work out if jobIds in embeddable input contain group names
2. Create an Anomaly Explorer URL, populating both job IDs and group names if present 
3. Update the Anomaly Explorer state code to resolve group names from the URL. The page fetches jobs upfront anyway, so we shouldn't introduce any redundant calls

Reply:
I modified the implementation accordingly, removing redundant API calls on the dashboards page and moving the logic to `useJobSelection` without introducing any additional API calls.
In: [#2b2cd2c](https://github.com/elastic/kibana/pull/198256/commits/2b2cd2c1b76611ef6900ac88f796b33690df26bb)
and improved a little bit to handle invalid Ids as well, in: [#a8a9eda](https://github.com/elastic/kibana/pull/198256/commits/a8a9eda572d0611a26ac841735715e04cfb3d93a)